tracer.go

```diff
@@ -206,6 +208,8 @@ func (t *Tracer) startSpanWithOptions(
 		ctx.traceID.Low = t.randomID()
 		if t.options.gen128Bit {
 			ctx.traceID.High = t.randomID()
+		} else if t.options.highTraceIDGenerator != nil {
```

Comment:
I don't think `else if` is appropriate - if we're not asked to generate 128bit IDs, then there's no need to invoke High ID generator even if it is provided. On the other hand, if `gen128Bit == true` but the external generator is not provided, it needs to be defaulted to `tracer.randomID`

Reply:
Fair enough. I think I just didn't think about the change in logic after changing the option to a function. This was originally checking the `ConsistentHighTraceID` flag.